pkg/controllers/disruption/controller.go

```diff
@@ -126,8 +109,10 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 
 	// Karpenter taints nodes with a karpenter.sh/disruption taint as part of the disruption process
 	// while it progresses in memory. If Karpenter restarts during a disruption action, some nodes can be left tainted.
-	// Idempotently remove this taint from candidates before continuing.
-	if err := c.requireNoScheduleTaint(ctx, false, c.cluster.Nodes()...); err != nil {
+	// Idempotently remove this taint from candidates that are not in the orchestration queue before continuing.
+	if err := state.RequireNoScheduleTaint(ctx, c.kubeClient, false, lo.Filter(c.cluster.Nodes(), func(s *state.StateNode, _ int) bool {
+		return !c.queue.HasAny(s.ProviderID())
+	})...); err != nil {
 		return reconcile.Result{}, fmt.Errorf("removing taint from nodes, %w", err)
 	}
```

Comment:
So this is essentially a GC loop in case we error out elsewhere and leak a bunch of tainted nodes. 

What if we crashed and the nodes are tainted (persisted) but the queue is empty?